File: Web-Pi/ajax/get_live.php
<?php
include('ajaxroot.php');
$dir = $ajaxroot.'userdata/cache';
$dir = $ajaxroot.'pi/projets/MyChronoGPS/cache';

$result = array();
$locfile = 'LIVE';
if (is_file($dir.'/'.$locfile)) {
	$handle = @fopen($dir.'/'.$locfile, "r");
	if ($handle) {
		while ($info = fgets($handle)) {
			array_push($result,$info);
		}
		fclose($handle);
	}
}
else {
	$info = new stdClass();
	$info->msgerr = 'fichier '.$dir.'/'.$locfile.' non trouvé';
	array_push($result,$info);
}

if (count($result) == 0) {
	$info = new stdClass();
	$info->msgerr = 'pas de coordonnée trouvée';
	array_push($result,$info);
}
$json = json_encode($result);
echo $json
?>

File: Web-Pi/ajax/get_live.py
#!/usr/bin/env python3
# -*- coding: UTF-8 -*-
from MyChronoGPS_Paths import Paths
Path = Paths();
dir = Path.pathdata+"/cache"
import glob 
import os
import json
print('Content-Type: text-plain; charset=utf-8\r\n\r\n')

result = []
locfile = 'LIVE'
fic = dir+'/'+locfile
if os.path.isfile(fic): 
    FD = open(fic, 'r')
    info = json.loads(FD.read())
    #rec = FD.read()
    #info = dict()
    #info["live"] = rec
    result.append(info)
    FD.close()
else:
    info = dict()
    info["msgerr"] = 'file '+fic+' not found'
    result.append(info)
if len(result) == 0:
    info = dict()
    info["msgerr"] = 'no coord found'
    result.append(info)

json_str = json.dumps(result)
print(json_str)
